plugin/iamutil/iam_resources.go

```diff
@@ -216,10 +216,16 @@ type ServiceConfig struct {
 	ServicePath string
 }
 
-func (r *iamResourceImpl) SetIamPolicyRequest(p *Policy) (*http.Request, error) {
-	data := struct {
-		Policy *Policy `json:"policy,omitempty"`
-	}{Policy: p}
+func (r *iamResourceImpl) SetIamPolicyRequest(p *Policy) (req *http.Request, err error) {
+	var data interface{}
+	switch strings.ToLower(r.config.Service.Name) {
+	case "storage":
```

Comment:
Could you add a comment here describing the difference between these to cases?

Reply:
done